hoomd/hpmc/shape_move.py

```diff
@@ -0,0 +1,322 @@
+# Copyright (c) 2009-2022 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+import hoomd
+from hoomd.operation import _HOOMDBaseObject
+from . import _hpmc
+from hoomd.hpmc import integrate
+from hoomd.data.parameterdicts import ParameterDict
+from hoomd.logging import log
+import numpy
+
+
+class ShapeMove(_HOOMDBaseObject):
+    """Base class for all shape moves.
+
+    A shape move is used as an argument to hoomd.hpmc.update.Shape to specify
+    how to alter shape definitions
+
+    Note:
+        This class should not be instantiated by users. The class can be used
+        for `isinstance` or `issubclass` checks.
+    """
+
+    def _attach(self):
+        self._apply_param_dict()
+        self._apply_typeparam_dict(self._cpp_obj, self._simulation)
+
+
+class Callback(_HOOMDBaseObject):
+    """Base class for callbacks used in Python shape moves.
+
+    Note:
+        This class should not be instantiated by users. User-defined callbacks
+        should inherit from this class, defining a __call__ method that takes
+        a list of floats as an input and returns a shape definition
+
+    Examples::
+
+        class ExampleCallback(hoomd.hpmc.shape_move.Callback):
+            def __call__(self, params):
+                # do something with params and define verts
+                return hoomd.hpmc._hpmc.PolyhedronVertices(verts)
```

Comment:
User-facing APIs should not access `_hpmc` directly. Modify the interface to accept `dict` matching the specification used in the `ConvexPolyhedron` `shape` parameter and convert to `PolyhedronVertices` on the C++ side.